mongodbatlas/resource_mongodbatlas_federated_settings_organization_role_mapping.go

```diff
@@ -68,6 +74,32 @@ func resourceMongoDBAtlasFederatedSettingsOrganizationRoleMapping() *schema.Reso
 	}
 }
 
+func validateRoleName() schema.SchemaValidateDiagFunc {
+	return func(v any, p cty.Path) diag.Diagnostics {
+		value := v.(string)
+		expected := []string{"ORG_OWNER", "ORG_GROUP_CREATOR", "ORG_BILLING_ADMIN", "ORG_READ_ONLY", "ORG_MEMBER", "GROUP_OWNER", "GROUP_CLUSTER_MANAGER", "GROUP_DATA_ACCESS_ADMIN", "GROUP_DATA_ACCESS_READ_WRITE", "GROUP_DATA_ACCESS_READ_ONLY", "GROUP_READ_ONLY", "GROUP_SEARCH_INDEX_EDITOR"}
```

Comment:
This list is missing some roles (see [createRoleMapping](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/Federated-Authentication/operation/createRoleMapping))

`"GLOBAL_AUTOMATION_ADMIN" "GLOBAL_BACKUP_ADMIN" "GLOBAL_METERING_USER" "GLOBAL_METRICS_INTERNAL_USER" "GLOBAL_MONITORING_ADMIN" "GLOBAL_OWNER" "GLOBAL_READ_ONLY" "GLOBAL_USER_ADMIN" "GLOBAL_USER_READ_ONLY" "GLOBAL_ACCOUNT_SUSPENSION_ADMIN" "GLOBAL_BILLING_ADMIN" "GLOBAL_LEGAL_ADMIN" "GLOBAL_FEATURE_FLAG_ADMIN" "GLOBAL_ATLAS_TSE" "GLOBAL_ATLAS_OPERATOR" "GLOBAL_ATLAS_ADMIN" "GLOBAL_STITCH_ADMIN" "GLOBAL_CHARTS_ADMIN" "GLOBAL_EXPERIMENT_ASSIGNMENT_USER" "GLOBAL_STITCH_INTERNAL_ADMIN" "GLOBAL_BAAS_FEATURE_ADMIN" "GLOBAL_SECURITY_ADMIN" "GLOBAL_QUERY_ENGINE_INTERNAL_ADMIN" "GLOBAL_PROACTIVE_SUPPORT_ADMIN" "GLOBAL_INFRASTRUCTURE_INTERNAL_ADMIN" "GLOBAL_SALESFORCE_ADMIN" "GLOBAL_SALESFORCE_READ_ONLY" "GLOBAL_APP_SERVICES_CLUSTER_DEBUG_DATA_ACCESS" "ORG_MEMBER" "ORG_READ_ONLY" "ORG_BILLING_ADMIN" "ORG_GROUP_CREATOR" "ORG_OWNER" "ORG_TEAM_MEMBERS_ADMIN" "GROUP_AUTOMATION_ADMIN" "GROUP_BACKUP_ADMIN" "GROUP_MONITORING_ADMIN" "GROUP_OWNER" "GROUP_READ_ONLY" "GROUP_USER_ADMIN" "GROUP_BILLING_ADMIN" "GROUP_DATA_ACCESS_ADMIN" "GROUP_DATA_ACCESS_READ_ONLY" "GROUP_DATA_ACCESS_READ_WRITE" "GROUP_CHARTS_ADMIN" "GROUP_CLUSTER_MANAGER" "GROUP_SEARCH_INDEX_EDITOR"`

Reply:
agree, @martinstibbe is it possible to implement that for this PR?

Reply:
@andreaangiolillo, thank you. I totally agree, hardcore is not the best solution, but I did't find how to get the roles list.

Reply:
@Zuhairahmed we have a policy of not hard coding values like this as it means the provider can end up behind the platform.  If we get an API endpoint from which to update them we could add a validation but that does not exist.  This PR should not be accepted.

Reply:
Also just FYI the global ones do not apply externally.